test_collections/matter/sdk_tests/support/python_testing/models/python_test_parser.py

```diff
@@ -192,39 +193,32 @@ def __get_method_by_name(
     return next((m for m in methods if name in m.name), None)
 
 
-def __retrieve_steps(method: FunctionDefType) -> List[MatterTestStep]:
-    python_steps: List[MatterTestStep] = []
-
+def __retrieve_is_commissioning(method: FunctionDefType) -> bool:
     steps_body = __retrieve_return_body(method, ast.List)
     if not steps_body:
         return []
 
-    for step in steps_body.value.elts:
-        try:
-            arg_is_commissioning = False
-
-            if (
-                step.keywords
-                and "is_commissioning"
-                in step.keywords[KEYWORD_IS_COMISSIONING_INDEX].arg
-            ):
-                arg_is_commissioning = step.keywords[
-                    KEYWORD_IS_COMISSIONING_INDEX
-                ].value.value
-
-            step_name = step.args[ARG_STEP_DESCRIPTION_INDEX].value
-            parsed_step_name = step_name
-        except Exception as e:
-            logger.error(f"Error while parsing step from {method.name}, Error:{str(e)}")
-            parsed_step_name = "UNABLE TO PARSE TEST STEP NAME"
-
-        python_steps.append(
-            MatterTestStep(
-                label=parsed_step_name, is_commissioning=arg_is_commissioning
-            )
+    for element in steps_body.value.elts:
+        step = element
+        break
```

Comment:
Is this just getting the first element of the list? Why use a `for` for that?

Reply:
Please take a second look on the refactored code